File: python/quicksort.py
# QuickSort in Python
def quicksort(arr):
    if len(arr) <= 1:
        return arr
    pivot = arr[len(arr) // 2]
    less = [x for x in arr if x < pivot]
    equal = [x for x in arr if x == pivot]
    greater = [x for x in arr if x > pivot]
    return quicksort(less) + equal + quicksort(greater)

# Example array
array = [3, 6, 8, 10, 1, 2, 1]

# Sort the array
sorted_array = quicksort(array)
sorted_array

File: R/quicksort.r
quicksort <- function(arr) {
  if (length(arr) <= 1) {
    return(arr)
  } else {
    pivot <- arr[floor(length(arr) / 2)]
    left <- arr[arr < pivot]
    middle <- arr[arr == pivot]
    right <- arr[arr > pivot]
    return(c(quicksort(left), middle, quicksort(right)))
  }
}

# Example usage:
example_array <- c(3, 6, 8, 10, 1, 2, 1)
sorted_array <- quicksort(example_array)
print(sorted_array)
